src/corona_hakab_model/population_analysis/population_analysis.py

```diff
@@ -0,0 +1,81 @@
+import math
+import pickle
+
+import numpy as np
+from generation.circles_generator import PopulationData
+from generation.connection_types import ConnectionTypes
+from generation.matrix_generator import MatrixData
+from matplotlib import pyplot as plt
+
+
+class PopulationAnylzer:
+    """
+    this module is in charge of plotting statistics about the generation stage, and specifically the population
+    """
+
+    __slots__ = "population_data", "matrix_data"
+
+    def __init__(self, population_data_path, matrix_data_path=None):
+
+        # importing population data from path
+        try:
+            with open(population_data_path, "rb") as population_data:
+                self.population_data: PopulationData = pickle.load(population_data)
```

Comment:
instead of using pickle, use the MatrixGenerator and CirclesGenerator's import function.